tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -382,6 +391,11 @@ std::vector<std::unique_ptr<TChain>> MakeFriends(const ROOT::TreeUtils::RFriendI
             frChain->Add((thisFriendFileNames[j] + "?#" + thisFriendChainSubNames[j]).c_str(), thisFriendEntries[j]);
          }
       }
+
+      const auto &indexInfo = finfo.fTreeIndexInfos[i];
```

Comment:
To make sure that `finfo.fTreeIndexInfos[i]` always exists, we should probably also add an empty `TreeIndexInfo` in all the `RFriendInfo::AddFriend` overloads

Reply:
This is probably not relevant for `TTreeProcessorMT` but it would be relevant for `RLoopManager(RDatasetSpec &&spec)`. See https://github.com/root-project/root/pull/12262 , tests were failing due to the missing addition of the friend entries in the `RFriendInfo::AddFriend` overloads. This would probably mean that as soon as that PR is merged also this one would start showing the same failures

Reply:
should be done